src/safeposix/syscalls/sys_calls.rs

```diff
@@ -2,22 +2,62 @@
 
 use crate::interface;
 use crate::safeposix::cage::{CAGE_TABLE, Cage, FileDescriptor::*};
+use crate::safeposix::filesystem::{FS_METADATA, Inode};
 
 use super::sys_constants::*;
 
 impl Cage {
   pub fn fork_syscall(&self, child_cageid: u64) -> i32 {
-    CAGE_TABLE.write().unwrap().insert(child_cageid, interface::RustRfc::new(Cage {cageid: child_cageid, cwd: self.cwd.clone(), parent: self.cageid, filedescriptortable: interface::RustLock::new(self.filedescriptortable.read().unwrap().clone())}));
+    let mut cagetab = CAGE_TABLE.write().unwrap();
```

Comment:
There's a couple cases where being more descriptive in the variable names just makes sense to do. No real sense cutting of the "le" here.